atomic_reactor/plugins/gather_builds_metadata.py

```diff
@@ -54,6 +57,56 @@ def _generate_build_log_output(self, platform: str, buildroot_id: int) -> Option
         metadata['arch'] = platform
         return Output(metadata=metadata, filename=str(build_log_file))
 
+    def _get_hostname_for_platform(self, platform: str) -> str:
+        task_results = self.workflow.osbs.get_task_results(self.workflow.pipeline_run_name)
+        task_platform = platform.replace('_', '-')
+        build_host = ""
+
+        for task_name in task_results:
+            if not (task_name.startswith('binary-container-build') and task_platform in task_name):
+                continue
+
+            if 'task_result' not in task_results[task_name]:
+                raise RuntimeError(f"task_results is missing from: {task_name}")
+
+            build_host = json.loads(task_results[task_name]['task_result'])
+            break
+
+        if not build_host:
+            raise RuntimeError(f"unable to find build host for platform: {platform}")
+        return build_host
+
+    def _get_build_rpms(self, platform: str, build_host: str):
+        remote_host = None
+
+        remote_host_pools = self.workflow.conf.remote_hosts.get("pools", {})
+        slots_dir = self.workflow.conf.remote_hosts.get("slots_dir")
+        platform_config = remote_host_pools.get(platform)
+
+        if not platform_config:
+            raise RuntimeError(f"unable to find remote hosts for platform: {platform}")
+
+        for hostname, attr in platform_config.items():
+            if build_host != hostname:
+                continue
+
+            remote_host = RemoteHost(
+                hostname=hostname, username=attr["username"], ssh_keyfile=attr["auth"],
+                slots=attr.get("slots", 1), socket_path=attr["socket_path"], slots_dir=slots_dir
+            )
+            break
```

Comment:
this whole for loop seems redundant to me. Can't we simply do something like this?
```
if build_host in platform_config: # maybe even this check is redundant
    attr = platform_config[build_host]
    remote_host = RemoteHost(
                hostname=build_host, username=attr["username"], ssh_keyfile=attr["auth"],
                slots=attr.get("slots", 1), socket_path=attr["socket_path"], slots_dir=slots_dir
            )

```
This way we avoid the for loop, ugly continue and break commands.

Reply:
ok